src/character.cpp

```diff
@@ -484,26 +484,46 @@ VisitResponse Character::visit_items( const std::function<VisitResponse( item& )
     return inv.visit_items( func );
 }
 
-VisitResponse Character::visit_items( const std::function<VisitResponse( const item& )>& func ) const
+VisitResponse Character::visit_items( const std::function<VisitResponse( const item *, const item * )>& func ) const
```

Comment:
Looks like you're overly specifying the visit function for that one usage.
In most invocations the second argument will be unused. There are 64 matches for "visit_items" on this page, but only one of them actually makes use of the second argument.

Can't it be done in some other way? At the very least by adding the two argument function as an overload rather than making it the only way to do it.